### DISCUSSION TITLE
Getting Started - How can i pass the cached_image_data by parameter request for a Rails API Only?

### DISCUSSION THREAD

**Discussion by Thiteago** [shrinerb/shrine#709] [Q&A]
Hey guys, i'm having a little trouble to figure out how the Shrine works.

I'm following the Getting Started from the documentation and follow the instructions step by step.

But there is a part of the examples who says about a hidden field on a form, but i'm kinda confuse, because my rails app is only API mode, and i don't know if Shrine keeps working on this, or if i need to pass anything more on my request  from external app.

For testing i'm sending an post request with a form data as body content , and on it, i have uploaded some images, but when i try to create the new record, he fails, gives me that error

```
#<ActionDispatch::Http::UploadedFile:0x00007f77e25bc960 
@tempfile=#<Tempfile:/tmp/RackMultipart20241015-358962-wows2a.jpg>, @original_filename=\"1_rIkmavUeqyRySwlQdA9kKg (Medium).jpg\", 
@content_type=\"image/jpeg\", 
@headers=\"Content-Disposition: form-data; name=\\\"tour_image-2\\\"; filename=\\\"1_rIkmavUeqyRySwlQdA9kKg (Medium).jpg\\\"\\r\\nContent-Type: image/jpeg\\r\\n\">] 

is not a valid IO object (it doesn't respond to #read, #eof?, #rewind, #close)"

```

And this is my actual code for creation

```ruby
def create_tour
      if(@params[:tour_cover_image].present?)
        tour_images = @params.select{ |key, _| key.to_s.start_with?('tour_image-') }
        @tour_params[:cover_image] = @params[:tour_cover_image]
        @tour_params[:images] = tour_images.values
        tour = Tour.new(@tour_params)
        tour.save!
      else
        tour = Tour.new(@tour_params.except(:images, :cover_image))
        tour.save!
      end
      tour
    end
```

So tha's it, i hope i can make it clear because english is not my primary language, so sorry about any mistakes here, and i appreciate any help.

**Comment by janko** [comment]
What version of Rails are you on?

**Reply by Thiteago** [reply]
I'm using rails 7
